src/client_side.h

```diff
@@ -148,6 +148,7 @@ class ConnStateData:
         bool reading = false; ///< we are monitoring for peer connection closure
         bool zeroReply = false; ///< server closed w/o response (ERR_ZERO_SIZE_OBJECT)
         bool peerAccessDenied = false; ///< cache_peer_access denied pinned connection reuse
+        bool borrowed = false; ///< this pinned connection was borrowed at least once
```

Comment:
There is a case when the idle pinned connection should not be closed during cache_peer removal:  after CONNECT but before GET request (ssl-bumped connection, ConnStateData::httpsPeeked()). That is how Squid worked before these changes. I could not find any existing flag/state for that purpose - so I had to add a new one, that is set in ConnStateData::borrowPinnedConnection().